src/coreclr/vm/frames.h

```diff
@@ -2097,7 +2097,22 @@ class HijackFrame : public Frame
     }
 
     virtual void UpdateRegDisplay(const PREGDISPLAY, bool updateFloats = false);
+
+#ifdef TARGET_X86
+    // On x86 we need to specialcase return values
     virtual void GcScanRoots(promote_func *fn, ScanContext* sc);
+#else
+    // On non-x86 platforms HijackFrame is just a more compact form of a resumable
+    // frame with main difference that OnHijackTripThread captures just the registers
+    // that can possibly contain GC roots.
+    // The regular reporting of a top frame will report everything that is live
+    // after the call as specified in GC info, thus we do not need to worry about
+    // return values.
+    virtual unsigned GetFrameAttribs() {
+        LIMITED_METHOD_DAC_CONTRACT;
+        return FRAME_ATTR_RESUMABLE;    // Treat the next frame as the top frame.
```

Comment:
The key change. on non-x86 we can treat the thread self-interrupted synchronously by a hijack the same as a thread interrupted asynchronously by a signal. The difference is the captured context. 

In theory the hijack could capture thr entire context as the OS does, but that would be fairly redundant, so we capture only what we need.

There is a small catch here. If in the future we need to report more registers (say we learn to return 3 GC pointers), we would need to adjust `OnHijackTripThread` and `HijackFrame::UpdateRegDisplay` accordingly. 
If we captured the entire context as OS interrupts do, we would not have to adjust. Things would "just work". 
I thought about this and, considering that a big portion of context cannot possibly contain GC pointers, not capturing entire context seems the right approach.